### PR TITLE
Merges from ironpython2

### PR DESCRIPTION
**Pull request by slide**

### CODE REVIEW

**Comment by slide** [comment]
This fixes #248

File: Src/IronPython/Runtime/Operations/IntOps.Generated.cs [hunk 1]
```diff
@@ -98,6 +99,9 @@ public static partial class SByteOps {
             return (x != 0);
         }
 
+        public static string __repr__(SByte x) {
```
Comment (slozier) [inline Src/IronPython/Runtime/Operations/IntOps.Generated.cs]: Can you do the corresponding change in the generator script?

**Comment by slide** [comment]
This is not a specific set of PR's. I basically did a diff between the source of IPy3 and IPy2 and merged some of the fixes in.

File: Src/IronPython/Runtime/Operations/IntOps.Generated.cs (hunk 1 above)
Comment (slide) [inline Src/IronPython/Runtime/Operations/IntOps.Generated.cs]: It was already there, I did that when I changed __nonzero__ to __bool__

File: Src/IronPython/Hosting/PythonCommandLine.cs [hunk 2]
```diff
@@ -386,7 +414,6 @@ public sealed class PythonCommandLine : CommandLine {
             } catch (ThreadAbortException tae) {
                 KeyboardInterruptException pki = tae.ExceptionState as KeyboardInterruptException;
                 if (pki != null) {
-                    Console.WriteLine(Language.FormatException(tae), Style.Error);
```
Comment (slide) [inline Src/IronPython/Hosting/PythonCommandLine.cs]: https://github.com/IronLanguages/main/commit/a7a522f3c4319e16183aff236854debe63193c8e

File: Src/IronPython/Hosting/PythonCommandLine.cs [hunk 3]
```diff
@@ -132,8 +132,14 @@ public sealed class PythonCommandLine : CommandLine {
...
             return result;
+        }
 
-
+        protected override int RunInteractiveLoop() {
+            var sys = Engine.GetSysModule();
+                        
+            sys.SetVariable("ps1", ">>> ");
+            sys.SetVariable("ps2", "... ");
+            return base.RunInteractiveLoop();
```
Comment (slide) [inline Src/IronPython/Hosting/PythonCommandLine.cs]: https://github.com/IronLanguages/main/commit/3f97f4053f6193917a7da8de2fddd1e095c3929a

File: Src/IronPython/Hosting/PythonCommandLine.cs [hunk 4]
```diff
@@ -225,6 +232,27 @@ public sealed class PythonCommandLine : CommandLine {
...
+                if (warnoptions != null && !string.IsNullOrEmpty(warnings)) {
+                    string[] warns = warnings.Split(',');
+                    foreach(string warn in warns) {
+                        warnoptions.Add(warn);
+                    }
+                }
+            }
+#endif
+        }
+
```
Comment (slide) [inline Src/IronPython/Hosting/PythonCommandLine.cs]: https://github.com/IronLanguages/main/commit/dc020d73a4cf6118beb44a5c1cf661ecfb038ef4

**Comment by slozier** [comment]
Covers: IronLanguages/main#1320

**Pull request merged**